File: beecrowd/2289.py
def to_binary_easy(n):
    return bin(n)[2:]

def to_binary(n):
    rest = [0]*64
    i = 0
    while(n!=0):
        rest[i] = n%2
        n //= 2
        i += 1
    return rest[::-1]

def hamming(str1, str2):
    c = 0
    for i in range(64):
        c += 0 if str1[i] == str2[i] else 1
    return c

while True:
    x, y = input().split(' ')
    if x == y == '0':
        break
    print(hamming(to_binary(int(x)), to_binary(int(y))))
    print(2**64-1)

File: beecrowd/2289.cpp
#include <iostream>

using namespace std;

int* to_binary(unsigned long long int n){
    int i = 0;
    int rest[64] = {0};
    for(int i = 0; n!=0; i++){
        rest[i] = n%2;
        n /= 2;
    }
    return rest;
}

int main(){

    unsigned long long int x, y;
    while(true){
        cin >> x >> y;
        if(!(x||y)) break;

        int bin_x[64] = {0};
        int bin_y[64] = {0};

        int i, dist = 0;
        // Transforma o primeiro em binário
        for(i = 0; x!=0; i++){
            bin_x[i] = x%2;
            x /= 2;
        }

        // Tranforma o segundo em binário
        for(i = 0; y!=0; i++){
            bin_y[i] = y%2;
            y /= 2;
        }

        // Hamming
        for(i = 0; i < 64; i++){
            dist += bin_x[i] != bin_y[i] ? 1 : 0;
        }

        cout << dist << endl;
    }

    return 0;
}
